src/Database/LSMTree/Internal/CRC32C.hs

```diff
@@ -102,6 +103,25 @@ hPutAllCRC32C fs h = go 0
         then return (written', crc')
         else go written' bs' crc'
 
+-- | This function makes sure that the whole /lazy/ 'BSL.ByteString' is written.
+hPutAllChunksCRC32C :: forall m h
+                    .  Monad m
+                    => HasFS m h
+                    -> Handle h
+                    -> BSL.ByteString
+                    -> CRC32C -> m (Word64, CRC32C)
+hPutAllChunksCRC32C fs h = \lbs crc ->
+    foldM (uncurry putChunk) (0, crc) (BSL.toChunks lbs)
+  where
+    putChunk :: Word64 -> CRC32C -> BS.ByteString -> m (Word64, CRC32C)
+    putChunk !written !crc !bs = do
+      (n, crc') <- hPutSomeCRC32C fs h bs crc
+      let bs'      = BS.drop (fromIntegral n) bs
+          written' = written + n
+      if BS.null bs'
+        then return (written', crc')
+        else putChunk written' crc' bs'
```

Comment:
Similar to how `hPutAll` for lazy bytestrings does it, you could reuse `hPutAllCRC32C` here
```suggestion
    foldM (uncurry putChunk) (0, crc) (BSL.toChunks lbs)
  where
    putChunk :: Word64 -> CRC32C -> BS.ByteString -> m (Word64, CRC32C)
    putChunk !written !crc !bs = do
      (n, crc') <- hPutAllCRC32C fs h bs crc
      pure $ (written + n, crc')
```

Reply:
Oh, absolutely. Thanks!